casa/Logging/LogSink.h

```diff
@@ -302,15 +300,15 @@ class LogSink : public LogSinkInterface
   static void createGlobalSink();
 
   //# Data members.
-  CountedPtr<LogSinkInterface> local_sink_p;
-  static CountedPtr<LsiIntermediate> * global_sink_p;
+  std::shared_ptr<LogSinkInterface> local_sink_p;
+  static std::shared_ptr<LsiIntermediate> * global_sink_p;
   static std::once_flag theirCallOnceFlag;
 
   // The following is a reference to the global sink. It is created to
   // ensure that the global sink is not destroyed before the last local
   // reference to it is destroyed. This can happen if you have a static
   // LogSink (or LogIO).
-  CountedPtr<LsiIntermediate> local_ref_to_global_p;
+  std::shared_ptr<LsiIntermediate> local_ref_to_global_p;
```

Comment:
I don't understand how "This can happen if you have a static LogSink (or LogIO)": The static member should be destroyed only at the end of the application, after all regular objects are already destroyed.

Reply:
I've changed the comment bit. I think the author meant that the order of destruction of global objects is not well defined, so if a LogSink object is used as a global object, the overall global LogSink might be deleted before it.